rfcs/convergence/slot-null-rendering.md

```diff
@@ -0,0 +1,232 @@
+    button: resolveShorthand(props.button, {
+      children: React.Fragment // 🚨 getSlots requires children
+    }),
+    icon: resolveShorthand(props.icon)
+  };
+
+  const { slots, slotProps } = getSlots(state, ['input']);
+
+  return (
+    <slots.root {...slotProps.root}>
+      <slots.button {...slotProps.button}>
+        <slots.icon {...slotProps.icon} />
+      </slots.button>
+    </slots.root>
+  );
+}
+```
+[Live Example](https://stackblitz.com/edit/react-ts-wwguyp?file=SlotAsParent.tsx)
+
+### Native elements without children
+
+e.g: An Icon that doesn't have children `<i class="icon"/>` (this cases is hypothetical, but quite possible, as we cannot verify it in converged components)
+
+## Detailed Design or Proposal
+
+Both options here proposed by the end are equivalent, the only difference is the impact on `ObjectShorthandProps` interface
+
+### Option 1: Make `shorthands` optional
+
+This solves this problem by verifying if the shorthand is `undefined` or not.
+By verifying if shorthand is `undefined` we can opt for null rendering without compromising the cases which native slots don't have children
+
+- `getSlots` should stop verifying null rendering by the existence of the property `children`, instead depending on the existence of the shorthand itself
+
+```tsx
+// react-utilities/src/compose/getSlots.ts
+// ....
+if (typedState[name] === undefined) {
+  slots[name] = nullRender;
+  continue;
+}
+```
+
+- Since some slots may have default props, verification must be done on shorthand declaration to ensure that those slots will be null rendered even presenting default props
+
+```diff
+export function resolveShorthand(
+  value,
+  defaultProps,
++  {optional = true}
++) {
++  // verify if shorthand is undefined
++  if (value === null || (value === undefined && optional)) {
++    return undefined;
++  }
+}
+```
+
+The problem with this approach is that the shorthand signature now will carry with it an `undefined` value, making overrides a little less elegant:
+
+```ts
+export const CustomAccordionHeader = React.forwardRef<HTMLElement, AccordionHeaderProps>((props, ref) => {
+  const state = useAccordionHeader(props, ref);
+
+  if (state.icon) {
+    state.icon.onClick = () => {
+      /**
+       * this is an override,
+       * if this was a hook than it would need
+       * to be declared before the conditional
+       **/
+    };
+  }
+
+  useAccordionHeaderStyles(state);
+  return renderAccordionHeader(state);
+});
+```
+
+In cases where we simply want to ensure that the slot should be rendered even when `shorthand` is `undefined` than `{optional: false}` should be provided
+
+```ts
+export const useState = (props, ref) => {
+  return {
+    ...props,
+    slot1: resolveShorthand(props.slot1), // undefined
+    slot2: resolveShorthand(props.slot2, {}, { optional: false }), // {}
+  };
+};
+```
+
+### Option 2: Symbol for null rendering
+
+This solves this problem by verifying if the shorthand has a special symbol or not.
```

Comment:
Can you include a demo of calling the API with the special symbol?

Reply:
The special symbol would only be used inside the methods declared to create `shorthands` and `slots`, meaning the methods `resolveShorthand` and `getSlots` from `react-utilities`, it should not leak for the component development itself.